crates/polars-ops/src/frame/join/iejoin/mod.rs

```diff
@@ -364,6 +364,15 @@ unsafe fn materialize_join(
     suffix: Option<PlSmallStr>,
 ) -> PolarsResult<DataFrame> {
     try_raise_keyboard_interrupt();
+    let mut sorted_left_row_idx = left_row_idx.sort(false).into_no_null_iter().peekable();
+    let left_idx_not_in_inner = (0..left.height() as u64).into_iter().filter(|left| {
+        while sorted_left_row_idx.next_if(|inner| inner < left).is_some() {}
+        sorted_left_row_idx.peek().map_or(true, |inner| left < inner)
+    });
+    let outer_left_row_idx: IdxCa = IdxCa::from_iter(left_idx_not_in_inner);
```

Comment:
`IdxCa::from_iter` expects an iterator over `Option<T::Native>` to allow for null values. You can instead use:
```suggestion
    let outer_left_row_idx: IdxCa = IdxCa::from_iter_values(PlSmallStr::EMPTY, left_idx_not_in_inner);
```